src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs

```diff
@@ -95,23 +100,38 @@ private void Commit(PresentationItem item, Model model, char? commitChar, Cancel
                         IgnoreTrimWhiteSpace = EditOptions.DefaultMinimalChange.DifferenceOptions.IgnoreTrimWhiteSpace
                     });
 
-                    // edit subject buffer (not view) because text changes are in terms of current document.
-                    using (var textEdit = this.SubjectBuffer.CreateEdit(editOptions, reiteratedVersionNumber: null, editTag: null))
+                    var textChangesInView = new List<TextChange>();
+
+                    var bufferGraph = new DisconnectedBufferGraph(this.SubjectBuffer, this.TextView.TextBuffer);
+                    foreach (var textChangeInBuffer in textChangesInSubjectBuffer)
+                    {
+                        var currentSpanInBuffer = new SnapshotSpan(
+                            this.SubjectBuffer.CurrentSnapshot,
+                            textChangeInBuffer.Span.ToSpan());
+
+                        var viewSpan = bufferGraph.GetSubjectBufferTextSpanInViewBuffer(
+                            currentSpanInBuffer.Span.ToTextSpan());
+
+                        textChangesInView.Add(new TextChange(viewSpan.TextSpan, textChangeInBuffer.NewText));
+                    }
+
+                    // Note: we currently create the edit on the textview's buffer.  This is 
+                    // necessary as our own 
```

Comment:
The handwriting just trails off at that point.  Do you think cyrus is in trouble?